sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -281,7 +281,7 @@ export class BatchingReceiverLite {
 
       if (receiver == null) {
         // (was somehow closed in between the init() and the return)
-        return [];
+        throw new ServiceBusError("Link closed before receiving.", "GeneralError");
```

Comment:
Nit for clarity:
```suggestion
        throw new ServiceBusError("Link closed before receiving messages.", "GeneralError");
```
Is this considered a retryable error?

Reply:
It is but, in the session case, it could fail if the session is already locked. 

I can catch and rethrow as a "SessionLockLost" error but it felt like it wasn't quite worth it because the next receive (after the failure) will report that error if they fail to get the session or it won't if they actually get it back.